pyseeyou/locales.py

```diff
@@ -48,60 +80,830 @@ def get_parts_of_num(num_string):
+
+cardinals["ses"] = cardinal_bm
+
+cardinals["sg"] = cardinal_bm
+
+cardinals["sh"] = cardinal_bs
+
+
+def cardinal_shi(n, i, v, w, f, t):
+    if i == 0 or n == 1:
+        return "one"
+    elif (n >= 2 and n <= 10):
+        return "few"
+
+    return "other"
+
+
+cardinals["shi"] = cardinal_shi
+
+
+def cardinal_si(n, i, v, w, f, t):
+    if n == 0 or n == 1 or i == 0 and f == 1:
+        return "one"
+
+    return "other"
+
+
+cardinals["si"] = cardinal_si
+
+cardinals["sk"] = cardinal_cs
+
+
+def cardinal_sl(n, i, v, w, f, t):
+    i100 = i % 100
+    if v == 0 and i100 == 1:
+        return "one"
+    elif v == 0 and i100 == 2:
+        return "two"
+    elif v == 0 and (i100 >= 3 and i100 <= 4) or v != 0:
+        return "few"
+
+    return "other"
+
+
+cardinals["sl"] = cardinal_sl
+
+cardinals["sma"] = cardinal_iu
+
+cardinals["smi"] = cardinal_iu
+
+cardinals["smj"] = cardinal_iu
+
+cardinals["smn"] = cardinal_iu
+
+cardinals["sms"] = cardinal_iu
+
+cardinals["sn"] = cardinal_af
+
+cardinals["so"] = cardinal_af
+
+cardinals["sq"] = cardinal_af
+
+cardinals["sr"] = cardinal_bs
+
+cardinals["ss"] = cardinal_af
+
+cardinals["ssy"] = cardinal_af
+
+cardinals["st"] = cardinal_af
+
+cardinals["sv"] = cardinal_ast
+
+cardinals["sw"] = cardinal_ast
+
+cardinals["syr"] = cardinal_af
+
+cardinals["ta"] = cardinal_af
+
+cardinals["te"] = cardinal_af
+
+cardinals["teo"] = cardinal_af
+
+cardinals["th"] = cardinal_bm
+
+cardinals["ti"] = cardinal_ak
+
+cardinals["tig"] = cardinal_af
+
+cardinals["tk"] = cardinal_af
+
+cardinals["tl"] = cardinal_fil
+
+cardinals["tn"] = cardinal_af
+
+cardinals["to"] = cardinal_bm
+
+cardinals["tr"] = cardinal_af
+
+cardinals["ts"] = cardinal_af
+
+
+def cardinal_tzm(n, i, v, w, f, t):
+    if (n >= 0 and n <= 1) or (n >= 11 and n <= 99):
+        return "one"
+
+    return "other"
+
+
+cardinals["tzm"] = cardinal_tzm
+
+cardinals["ug"] = cardinal_af
+
+cardinals["uk"] = cardinal_ru
+
+cardinals["ur"] = cardinal_ast
+
+cardinals["uz"] = cardinal_af
+
+cardinals["ve"] = cardinal_af
+
+cardinals["vi"] = cardinal_bm
+
+cardinals["vo"] = cardinal_af
+
+cardinals["vun"] = cardinal_af
+
+cardinals["wa"] = cardinal_ak
+
+cardinals["wae"] = cardinal_af
+
+cardinals["wo"] = cardinal_bm
+
+cardinals["xh"] = cardinal_af
 
-    return 'other'
+cardinals["xog"] = cardinal_af
 
+cardinals["yi"] = cardinal_ast
 
-def other(n, i, v, w, f, t):
-    return 'other'
+cardinals["yo"] = cardinal_bm
 
+cardinals["yue"] = cardinal_bm
 
-def spanish(n, i, v, w, f, t):
-    if i == 1 and w == 0:
-        return 'one'
-    return 'other'
+cardinals["zh"] = cardinal_bm
 
+cardinals["zu"] = cardinal_am
 
-LOCALE_FUNCTIONS = {
-    'cs': czech,
-    'de': one_or_other,
-    'en': one_or_other,
-    'fr': french,
-    'it': one_or_other,
-    'ja': other,
-    'nl': one_or_other,
-    'pt': brazilian_pt,
-    'pt_PT': one_or_other,
-    'zh': other,
-    'es': spanish
-}
+# =====================
+# END CODE BY GENERATOR
+# =====================
```

Comment:
This might be slight bikeshedding, but I think the naming of the functions here is quite confusing. 

The first locale (alphabetically) that has a set of operations on the parts of a number is then referred to by other locales that perform those same operations. As it stood before, for any cardinal functions that were used by more than one locale, an attempt was made to name them slightly more generically, however here we have: 

    cardinals["zu"] = cardinal_am

which I think isn't the best to instantly understand or maintain. I understand this code was auto-generated, but there's nothing to say we can't clean it up slightly :)

Also, I think creating a single `LOCALE_FUNCTIONS` or `CARDINALS` at the end of the file is slightly cleaner to look at than having lines assign functions to the dictionary throughout the file.